### PR TITLE
implementing high performance advanced algorithms in order to maintai…

### LINKED ISSUE #275: Cannot read the text
Cannot read the text and cannot resize the window, to make this visible:
Bildschirmfoto 2020-03-29 um 18 28 39

### PR DESCRIPTION
**Pull request by anLeid**
…n proper user experience

ideas and concept based on the work of jotoh

Fixes #275

### CODE REVIEW

**Review by DorianRec — DISMISSED** [review]
Bildschirmfoto 2020-03-30 um 01 37 38

File: src/main/java/com/treasure/hunt/view/settings/SettingsController.java [hunk 1]
```diff
@@ -56,6 +56,8 @@ private static void addDescription(VBox vBox, Field field) {
 
         if (!description.isEmpty()) {
             Label descriptionLabel = new Label(description);
+            descriptionLabel.setMinHeight(Double.NEGATIVE_INFINITY);
+            descriptionLabel.setWrapText(true);
             descriptionLabel.getStyleClass().add("description");
             vBox.getChildren().add(descriptionLabel);
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/view/settings/SettingsController.java:92]: So  @jotoh98 this shold also be defined in the fxml but I guess it is too late now

Comment (1-alex98) [reply src/main/java/com/treasure/hunt/view/settings/SettingsController.java:92]: I'd just turn it off then

File: src/main/java/com/treasure/hunt/view/settings/SettingsWindow.java [hunk 2]
```diff
@@ -14,17 +15,16 @@
 @Slf4j
 public class SettingsWindow {
 
-    public static void show() throws Exception {
+    public static void show() {
         Stage stage = new Stage();
-        final FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/layout/settings/settings.fxml"));
-        Scene scene = new Scene(fxmlLoader.load());
+        final Widget<SettingsController, Region> settings = new Widget<>("/layout/settings/settings.fxml");
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/view/settings/SettingsWindow.java]: Why is that used here Widget was not intended for this use if u ask me, what was wong with the old code

**Review by 1-alex98 — DISMISSED** [review]

File: src/main/java/com/treasure/hunt/view/settings/SettingsController.java (hunk 1 above)
Comment (jotoh98) [reply src/main/java/com/treasure/hunt/view/settings/SettingsController.java:92]: These are generated dynamically depending on fields in class Settings and their annotations. If wanted, I could implement one label as fxml and load it every time, but it seems a bit unnecessary considering the upper part of the corresponding setting block is changed for each type of field.

File: src/main/java/com/treasure/hunt/view/settings/SettingsWindow.java (hunk 2 above)
Comment (jotoh98) [reply src/main/java/com/treasure/hunt/view/settings/SettingsWindow.java]: Widget is a class generic enough to be used in this way. I thought DRY code would fit this case better

**Review by DorianRec — APPROVED** [review]

**Pull request merged**